lib/auth/auth_with_roles_test.go

```diff
@@ -5516,6 +5518,270 @@ func TestListUnifiedResources_IncludeRequestable(t *testing.T) {
 	}
 }
 
+func TestCreateAccessRequestV2_oktaReadOnly(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	srv := newTestTLSServer(t)
+
+	// 1. Create Okta-originated app server in the backend.
+
+	searchableOktaApp := newTestAppServerV3(t, srv.Auth(), "serachable-okta-app", map[string]string{
+		"name":            "serachable-okta-app",
+		types.OriginLabel: types.OriginOkta,
+	})
+
+	// 2. Create a role allowing the Okta app (used for search_as_roles)
+
+	searchRole, err := CreateRole(ctx, srv.Auth(), "serach_role", types.RoleSpecV6{
+		Allow: types.RoleConditions{
+			AppLabels: types.Labels{
+				"name": {searchableOktaApp.GetName()},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	// 3. Create the user role allowing to search Okta app
+
+	aliceRole, err := CreateRole(ctx, srv.Auth(), "alice_role", types.RoleSpecV6{
+		Allow: types.RoleConditions{
+			Request: &types.AccessRequestConditions{
+				SearchAsRoles: []string{
+					searchRole.GetName(),
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	// 4. Create the user
+
+	alice, err := CreateUser(ctx, srv.Auth(), "alice", aliceRole)
+	require.NoError(t, err)
+
+	// 5. Create a client for the user
+
+	aliceClt, err := srv.NewClient(TestUser(alice.GetName()))
+	require.NoError(t, err)
+
+	// 6. Prepare access requests
+
+	testAccessRequests := []types.AccessRequest{
+		// requesting app
+		mustAccessRequest(t, alice.GetName(), types.RequestState_PENDING, srv.Clock().Now(), srv.Clock().Now().Add(time.Hour),
+			[]string{}, // roles
+			[]types.ResourceID{
+				mustResourceID(srv.ClusterName(), types.KindApp, searchableOktaApp.GetName()),
+			},
+		),
+		// requesting app_server
+		mustAccessRequest(t, alice.GetName(), types.RequestState_PENDING, srv.Clock().Now(), srv.Clock().Now().Add(time.Hour),
+			[]string{}, // roles
+			[]types.ResourceID{
+				mustResourceID(srv.ClusterName(), types.KindAppServer, searchableOktaApp.GetName()),
+			},
+		),
+	}
+
+	// 7. Run tests
```

Comment:
Do we really want to add a comment for trivial steps?

A good documentation comment should focus on why something is done, rather than what, especially when the what is obvious at first glance.

So I would not include comment for trivial steps like 
``
// created variable X
testAccessRequests := []types.AccessRequest{

// start test X
t.Run(
```

Reply:
Is it about only this comment or all the comments before that? To me those comments are helpful, they describe the test scenario. The code is expressive and that's good, but in some places you need to dig for this one line like `searchableOktaApp.GetName()`.

Out of context, the `// 7. Run tests` comment alone doesn't make much sense as you point out. But it visually marks the end of the last setup step `// 6. Prepare access requests`.